README.md

```diff
@@ -49,6 +49,118 @@ In addition, this proposal introduces a new type of class element that can be de
 
 This new element type can be used independently, and has its own semantics separate from usage with decorators. The reason it is included in this proposal is primarily because there are a number of use cases for decorators which require its semantics, since decorators can only replace an element with a corresponding element that has the same semantics. These use cases are common in the existing decorators ecosystem, demonstrating a need for the capabilities they provide.
 
+## Motivation
+
+You might be wondering "why do we need these at all?" Decorators are a powerful metaprogramming feature that can simplify code significantly, but can also feel "magical" in the sense that they hide details from the user, making what's going on under the hood harder to understand. Like all abstractions, in some cases decorators can become more trouble than they're worth.
+
+However, one of the main reasons decorators are still being pursued today, and _specifically_ the main reason class decorators are an important language feature, is that they fill a gap that exists in the ability to metaprogram in JavaScript.
+
+Consider the following functions:
+
+```js
+function logResult(fn) {
+  return function(...args) {
+    try {
+      const result = fn.call(this, ...args);
+      console.log(result);
+    } catch (e) {
+      console.error(result);
+      throw e;
+    }
+    return result;
```

Comment:
`result` is now out of scope in the `return result` statement, and in the `console.error(result)`.

I'd personally also move the `console.log` out of the `try` block, but that's less of a glaring issue.

```suggestion
    let result
    try {
      result = fn.call(this, ...args);
    } catch (e) {
      console.error(e);
      throw e;
    }
    console.log(result);
    return result;
```